### DISCUSSION TITLE
如何支持第三方系统调用lamp-cloud的接口

### DISCUSSION THREAD

**Discussion by zqb666** [zuihou/zuihou-admin-cloud#200] [02.提问] [answered]
我现在有个单独的业务系统采用的spring boot，想调用lamp-cloud的用户角色权限等相关的接口，比如调用localhost:8764/roleAuthority/1  （查询指定角色关联的菜单和资源）但是我发现调用的时候后台会执行SELECT id, authority_id, authority_type, role_id, create_time, created_by FROM c_role_authority WHERE (role_id = 1) AND c_role_authority.tenant_code = ''  这样的sql 缺少 租户编码值，这个租户编码值我要怎么传过去？
还有就是这个接口可以直接被外部系统 访问 是不是缺乏安全性？  这样外部系统没有任何用户认证就可以直接能访问了

**Comment by zuihou** [comment, marked as answer]
1. 参考： HeaderThreadLocalInterceptor , 通过拦截器或者手动设置 `ContextUtil.setTenantBasePoolName(basePoolName);` 参数
2. 参考： AuthenticationFilter ，配置uri权限

**Reply by zqb666** [reply]
/**
* 切换base库
*
* @param tenant
*/
public static void setTenantBasePoolName(Object tenant) {
set(ContextConstants.TENANT_BASE_POOL_NAME_HEADER, tenant);
}

这个方法不是用来切换库的吗？

**Reply by zuihou** [reply]
不同的版本， `tenant_code = ''` 取值的参数不同，可能是 `ContextUtil.getTenant` 或 `ContextUtil.getTenantBasePoolName ` ，具体看 `TenantLineInnerInterceptor` 拦截器的设置：
```
public Expression getTenantId() {
                return new LongValue(ContextUtil.getTenant());
            }
```

**Reply by zqb666** [reply]
好的 谢谢

**Reply by zqb666** [reply]
关于第二点的 AuthenticationFilter  具体是什么  是 spring security的类吗

**Comment by zuihou** [comment]
3.x 无 AuthenticationFilter,  由 lamp-security-sdk 通过注解控制 Controller 层权限

**Reply by zuihou** [reply]
UriSecurityPreAuthAspect
